lib/block-write-stream.ts

```diff
@@ -9,16 +9,22 @@ import { makeClassEmitProgressEvents } from './source-destination/progress';
 
 const debug = _debug('etcher:writer:block-write-stream');
 
+const MIN_CHUNK_SIZE = 512;
 const CHUNK_SIZE = 64 * 1024;
-const UPDATE_INTERVAL_MS = 500;
 const RETRY_BASE_TIMEOUT = 100;
 
 export class BlockWriteStream extends Writable {
 	private bytesWritten = 0;
 	private _firstBlocks: { position: number, buffer: Buffer }[] = [];
 
-	constructor(private fd: number, private chunkSize = CHUNK_SIZE, private maxRetries = 5) {
+	private _buffers: Buffer[] = [];
+	private _bytes = 0;
+
+	constructor(private fd: number, private blockSize = MIN_CHUNK_SIZE, private chunkSize = CHUNK_SIZE, private maxRetries = 5) {
 		super({ objectMode: true, highWaterMark: 1 });
+		this.blockSize = blockSize;
+		this.chunkSize = chunkSize;
+		this.maxRetries = maxRetries;
```

Comment:
That'd be a question for @zvin, but as far as I can see they were previously unused. Likely a leftover from porting this from Etcher master

Reply:
That's not the reason. Since they are declared `private` in the constructor they will be automatically be added on `this` so you have to remove those extra lines. https://www.typescriptlang.org/play/index.html#src=class%20A%20%7B%0D%0A%20%20%20%20constructor(private%20foo)%20%7B%0D%0A%0D%0A%20%20%20%7D%0D%0A%7D%0D%0A